pkg/interface/src/views/apps/links/components/LinkSubmit.tsx

```diff
@@ -0,0 +1,230 @@
+import { BaseInput, Box, Button, LoadingSpinner, Text } from "@tlon/indigo-react";
+import React, { useCallback, useState } from "react";
+import GlobalApi from "~/logic/api/global";
+import { useFileDrag } from "~/logic/lib/useDrag";
+import useS3 from "~/logic/lib/useS3";
+import { S3State } from "~/types";
+import SubmitDragger from "~/views/components/SubmitDragger";
+import { createPost } from "~/logic/api/graph";
+import { hasProvider } from "oembed-parser";
+
+interface LinkSubmitProps {
+  api: GlobalApi;
+  s3: S3State;
+  name: string;
+  ship: string;
+};
+
+const LinkSubmit = (props: LinkSubmitProps) => {
```

Comment:
seems a little long, wonder if it's maybe worth turning into a Formik form?

Reply:
I just figured out hooks, but I haven't quite grokked formik. If I try more than one new tech per PR my head begins to spin a little.